content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,615 @@
+[`view/1_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_renderer.js).
+
+Our first step is to write a "Hello World" like application, that interacts on a very basic level with the XRP Ledger 
+and displays some high-level information about the current ledger state on the screen. Nothing too fancy so far, as we 
+will take care of styling and GUI related coding in a later step:
+
+![Screenshot: Step 1, hello world equivalent](img/javascript-wallet-1.png)
+
+First, we will create an entrypoint for our application, for this we create the file `1_hello.js` with the following content:
+
+`1_hello.js`
+{{ include_code("_code-samples/build-a-wallet/desktop-js/1_hello.js", language="js") }}
+
+This code has two parts: one that creates the application window and one that calls the XRP Ledger API's [ledger method](ledger.html). The code then broadcasts an event with the API response as the payload. The frontend picks up this event and uses the payload to display the index of most recently validated ledger.
+
+To display our results to the user, we need to create the view components that we specified in the `createWindow()` function. For this, we will create a `view` folder and add the following files there:
+
+<!-- MULTICODE_BLOCK_START -->
+
+*view/1_preload.js*
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/1_preload.js", language="js") }}
+
+*view/1_hello.html*
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/1_hello.html", language="html") }}
+
+*view/1_renderer.js*
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/1_renderer.js", language="js") }}
+
+<!-- MULTICODE_BLOCK_END-->
+
+The file `view/1_preload.js` does the main wiring of the application. The file `view/1_hello.html` defines the template part of the view. The file `view/1_renderer.js` contains scripts used in the template; you could also have put these in the HTML file, but it's a good practice to separate them so the code is more readable.
+
+This example shows how to do Inter Process Communication (IPC) in Electron. Technically, JavaScript has no true parallel processes or threading, because it follows a single-threaded event-driven paradigm. Nonetheless Electron provides us with two IPC modules called `ipcMain` and `ipcRenderer`. We can roughly equate those two to a backend process and a frontend process when we think in terms of client-server applications. It works as follows:
+
+1. Create a function that enables the frontend to subscribe to backend events (in `view/1_preload.js`)
+2. Make the function available by preloading it (webPreferences.preload during window creation)
+3. Create a frontend view
+4. Use that function in the frontend (e.g. 1_renderer.js, loaded in 1_hello.html) to attach a callback that handles frontend updates when the event is dispatched
+5. Dispatch the event from the backend (e.g. appWindow.webContents.send('update-ledger-index', value))
```

Comment:
This section uses very action-oriented language and so may be confusing to the reader as they aren't sure if they should actually be following the steps. Instead, I'd recommend describing it as if it's happening to a 3rd party instead, like this:
```suggestion
1. We start by create a function that enables the frontend to subscribe to backend events (in `view/1_preload.js`)
2. Then we make the function available by preloading it (webPreferences.preload during window creation)
3. This creates a frontend view
4. On the frontend, we can then use that function to attach a callback that handles frontend updates when the event is dispatched (e.g. How `1_renderer.js` is loaded in `1_hello.html`)
5. Lastly, we dispatch the event from the backend (e.g. appWindow.webContents.send('update-ledger-index', value))
```